plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/NoShallowCloneHook.java

```diff
@@ -0,0 +1,26 @@
+package org.jenkins.tools.test.hook;
+
+import org.apache.commons.lang.StringUtils;
+import org.jenkins.tools.test.PluginCompatTester;
+import org.jenkins.tools.test.model.hook.PluginCompatTesterHookBeforeCheckout;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class NoShallowCloneHook extends PluginCompatTesterHookBeforeCheckout {
+
+
+    private static final List<String> NO_SHALLOW_PLUGIN = Arrays.asList("git", "git-client");
+
+    @Override
+    public Map<String, Object> action(Map<String, Object> moreInfo) throws Exception {
+        String pluginName = (String) moreInfo.get("pluginName");
+        // such plugin need commit history for some tests
```

Comment:
Which tests? This is not acceptable. We should immediately `@Ignore` them pending a rewrite. Then we can delete the `SHALLOW_CLONE` flag and simplify considerably.

Reply:
(CC @MarkEWaite)

Reply:
The git client plugin fails 64 of 2264 tests with a shallow, single branch clone.  That's not a huge number.  They can probably be made intelligent enough to skip the test when they detect that they are running from a shallow clone or from a single branch clone.  I'm not sure when I'll have the time to make that change, but I think it is a worthwhile change.

I don't want to delete those tests because they test functionality that I don't want to risk regressing in future releases.  Happy to have help from others

The git plugin fails 5 of 3161 tests with a shallow, single branch clone.  They should be easy enough to safeguard those 5 tests so that a shallow, single branch clone can be used with those two plugins being a special case.

Reply:
I've submitted https://github.com/jenkinsci/git-plugin/pull/1365 as a git plugin pull request to resolve the failing git plugin tests when running with a shallow, single branch clone.

Reply:
thanks @MarkEWaite

Reply:
So we should be able to make an analogous fix for `git-client`?

Reply:
> So we should be able to make an analogous fix for `git-client`?

Yes, but the 64 failing tests in the git client plugin will need more time to fix or refactor.

Reply:
Ack. Can we at least leave behind an issue in this repo to delete the `SHALLOW_CLONE` option and revert all the complexity this PR introduced to support it after `git-client` is fixed?